### DISCUSSION TITLE
env var in vercel.json

### DISCUSSION THREAD

**Discussion by cgarnier** [vercel/vercel#5821] [Ideas]
Hello,
Could be nice to be able to inject env var during the deployement or permit to `envsubst` (or npm `envsub`) during the build.

```
{
  "rewrites": [
    {
      "source": "/storage/:matchs*",
      "destination": "${STORAGE}:matchs"
    }
  ]
}
```

I tried to generate the `vercel.json` during the build with a template and a prebuild command but the generated file is just ignored.

**Comment by styfle** [comment]
You can't do this with `vercel.json` because its static but you can use `next.config.js` to write dynamic config if you're using Next.js

**Reply by cgarnier** [reply]
`envsubst < versel.tpl.json > vercel.json` generate a static valide vercel conf. What i m saying is that i cant do it during the vercel build process. It s boring because i cant use the github integration. It force me to use a CI to do this simple task.
